### ISSUE TITLE
JDBC "application_name" parameter does not work because its invalid

### ISSUE THREAD

**Issue by timveil** [cockroachdb/docs#4294]
Using the latest version of cockroach `2.1.4` and the postgres JDBC driver `42.2.5` there seems to be difficulty passing `application_name`.

This seems to be because the correct parameter as defined by ~postgres~ the JDBC driver (in contrast to the actual setting supported by the postgres server) is actually `ApplicationName` not `application_name` see: https://jdbc.postgresql.org/documentation/head/connect.html

Therefore, specifically for JDBC examples, places where we suggest this: 
```
jdbc:postgresql://host:5432/db?application_name=foo
```
should actually be this...
```
jdbc:postgresql://host:5432/db?ApplicationName=foo
```

i have seen the incorrect parameter in a number of official and unofficial examples i believe.

BTW, this impacts "App" filtering on the "Statements" screen in our UI.

**Comment by jseldess** [comment]
Thanks for reporting, @timveil!

**Comment by knz** [comment]
As I have discussed with Tim, this is specific to the pgJDBC driver. The problem does not exist with many other drivers that connect to postgres/crdb directly.

**Comment by lhirata** [comment]
@timveil - After searching our docs to find the offending parameter, I can't seem to find any mention of it. Does this seem correct to you? If not, could you point me to what you're referring to? Thanks!

**Comment by lhirata** [comment]
I searched again for this update and couldn't find it. Closing this issue - feel free to reopen if you think otherwise!

**Comment by jseldess** [comment]
It's mentioned here, @lhirata. Definitely a thing. https://www.cockroachlabs.com/docs/stable/connection-parameters.html#additional-connection-parameters

**Comment by timveil** [comment]
This is a tricky one... the postgres protocol specifies `application_name` therefore most other drivers work just fine when using this variable.  the JDBC postgres driver mentioned above deviates from this standard and requires `ApplicationName`.  this means anywhere we are showing a Java/JDBC example we need to take care to use `ApplicationName` not the other.  Again this only applies to JDBC examples in our docs.  They may very well have been updated since I filed the ticket but they were certainly there at some point.
